datalad/distribution/create_sibling.py

```diff
@@ -392,7 +393,14 @@ def _create_dataset_sibling(
 def _ls_remote_path(ssh, path):
     try:
         # yoh tried ls on mac
-        out, err = ssh("ls -A1 {}".format(sh_quote(path)))
+        # escape path explicitly with shlex.quote as 'sh' is a POSIX shell and
+        # sh_quote could decide to quote Windows-style
+        ls_cmd = "LC_ALL=C; export LC_ALL; /bin/ls -A1 {}".format(shlex.quote(path))
+        # TODO: Using sh_quote here is also flawed as it checks whether the
+        # *local* machine is Windows. Doesn't help if the remote we're ssh'ing in is Windows.
```

Comment:
so should we then just hardcode `""` wrapping since IIRC single quote wrapping (`'???'`) is not available on Windows?

Reply:
No, hard-coding doesn't work because the filename could contain quotes that need to be dealt with. Using `shlex.quote` or `sh_quote` handle this.

I don't have access to a Windows SSH host to test this, maybe someone else can check if this works for normal non-exotic files with quotes in the name.

Reply:
> I don't have access to a Windows SSH host to test this, maybe someone else can check if this works for normal non-exotic files with quotes in the name.

``"`` is considered an illegal character on Windows systems, one couldn't create a file with it in its name: https://learn.microsoft.com/en-us/windows/win32/fileio/naming-a-file

Reply:
... and single quotes are not a thing in `cmd.exe`